src/main/java/de/dennisguse/opentracks/AbstractListActivity.java

```diff
@@ -62,17 +62,15 @@ protected void deleteTracks(long[] trackIds) {
     @Override
     public void onConfirmDeleteDone(long[] trackIds) {
         boolean stopRecording = false;
-        if (trackIds.length == 1 && trackIds[0] == -1L) {
-            stopRecording = true;
-        } else {
-            long recordingTrackId = PreferencesUtils.getRecordingTrackId(this);
-            for (long trackId : trackIds) {
-                if (trackId == recordingTrackId) {
-                    stopRecording = true;
-                    break;
-                }
+
+        long recordingTrackId = PreferencesUtils.getRecordingTrackId(this);
+        for (long trackId : trackIds) {
+            if (trackId == recordingTrackId) {
+                stopRecording = true;
+                break;
             }
         }
+
         if (stopRecording) {
             getTrackRecordingServiceConnection().stopRecording(this, false);
         }
```

Comment:
I like the best this transition for `TrackDeleteActivity` start (below `startActivityForResult`):

`overridePendingTransition(android.R.anim.fade_in, android.R.anim.fade_out);`